crates/recorder/src/measure/counters.rs

```diff
@@ -1,181 +1,59 @@
+    cache_misses: Counter,
 }
 
 impl CounterMeasure {
-    cfg_if::cfg_if! {
-        if #[cfg(target_os = "linux")] {
-            pub fn new() -> Self {
-                use perfcnt::linux::HardwareEventType;
-                let precision = Precision::new(precision::Config::default()).expect("to create a precision object");
+    pub fn new() -> Self {
+        cfg_if::cfg_if! {
+            if #[cfg(target_os = "linux")] {
+                let mut group = Group::new().expect("Unable to create event group; try setting /proc/sys/kernel/perf_event_paranoid to 2 or below?");
                 Self {
-                    precision,
-                    start_time: None,
-                    cpu_cycles: linux::create_hardware_counter(HardwareEventType::CPUCycles),
-                    instructions_retired: linux::create_hardware_counter(HardwareEventType::Instructions),
-                    cache_accesses: linux::create_hardware_counter(HardwareEventType::CacheReferences),
-                    cache_misses: linux::create_hardware_counter(HardwareEventType::CacheMisses)
-                }
-            }
-        } else {
-            pub fn new() -> Self {
-                let precision = Precision::new(precision::Config::default()).expect("to create a precision object");
-                // This doesn't currently support performance counter use on non-linux targets.
-                // `perfcnt` doesn't feature-gate `linux` so explicitly provide None's here rather
-                // than calling possibly wildly incorrect syscalls on non-linux targets.
-                Self {
-                    precision,
-                    start_time: None,
-                    cpu_cycles: None,
-                    instructions_retired: None,
-                    cache_accesses: None,
-                    cache_misses: None,
+                    cpu_cycles: Builder::new().group(&mut group).kind(Hardware::CPU_CYCLES).build().expect("Unable to create CPU_CYCLES hardware counter. Does this system actually have such a counter? If it does, your kernel may not fully support this processor."),
+                    instructions_retired: Builder::new().group(&mut group).kind(Hardware::INSTRUCTIONS).build().expect("Unable to create INSTRUCTIONS hardware counter. Does this system actually have such a counter? If it does, your kernel may not fully support this processor."),
+                    cache_accesses: Builder::new().group(&mut group).kind(Hardware::CACHE_REFERENCES).build().expect("Unable to create CACHE_REFERENCES hardware counter. Does this system actually have such a counter? If it does, your kernel may not fully support this processor."),
+                    cache_misses: Builder::new().group(&mut group).kind(Hardware::CACHE_MISSES).build().expect("Unable to create CACHE_MISSES hardware counter. Does this system actually have such a counter? If it does, your kernel may not fully support this processor."),
+                    event_group: group,
```

Comment:
`rustfmt` won't split strings into multiple lines by default, but we can do that without introducing any extra whitespace like this:

```rust
    foo().expect(
        "long line \
         continued",
    );
```

The leading whitespace on the new line of the string is ignored.

Doing this would make these lines easier to read.